File: src/main/java/camera/Player.java
package camera;

import javafx.scene.Node;
import javafx.scene.image.Image;

abstract class Player
{
    abstract int getFps();

    abstract void play();

    abstract void stop();

    abstract void makeScreenshot(String path);

    abstract Image getScreenshot();

    abstract Node getViewNode();

    abstract void setSource(String uri);
}

File: src/main/java/camera/Player.kt
package camera

import javafx.scene.Node
import javafx.scene.image.Image

internal abstract class Player
{
    internal abstract val fps: Int

    internal abstract val screenshot: Image

    internal abstract val viewNode: Node

    internal abstract fun play()

    internal abstract fun stop()

    internal abstract fun makeScreenshot(path: String)

    internal abstract fun setSource(uri: String?)
}
